dmsite/venv/src/Header/Header.jsx

```diff
@@ -26,32 +56,47 @@ class Header extends React.Component {
     });
   }
 
-  renderButton() {
+  renderSearchBar() {
     return (
-      <Button onClick={() => { this.props.history.push(`/search/${this.state.searchbarText}`); }}>Search</Button>
+      <Form inline>
+        <Input id="searchBar" type="search" className="ml-4" placeholder="Search Dataset" onChange={(event) => { this.updateSearchBarText(event); }} />
+        <InputGroupButtonDropdown addonType="append" isOpen={this.state.dropdownOpen} toggle={this.toggleDropDown}>
+          <DropdownToggle outline caret>
```

Comment:
65 - 75 has a lot of duplicate looking code. This should probably be refactored.